src/python/zquantum/core/measurement.py

```diff
@@ -347,31 +347,47 @@ def get_parities_from_measurements(
 
     # Count number of occurrences of bitstrings
     bitstring_frequencies = Counter(measurements)
+    bitstring_counts: np.ndarray = np.fromiter(
+        bitstring_frequencies.values(), dtype=int
+    )
 
     # Count parity occurrences
     values = []
     for _, term in enumerate(ising_operator.terms):
-        values.append([0, 0])
         marked_qubits = [op[0] for op in term]
-        for bitstring, count in bitstring_frequencies.items():
-            if check_parity(bitstring, marked_qubits):
-                values[-1][0] += count
-            else:
-                values[-1][1] += count
+        parity = check_parity_of_vector([*bitstring_frequencies.keys()], marked_qubits)
```

Comment:
[suggestion]I think it would be worthwhile to extract and call separately the method that transforms bitstrings into an array.